express/src/controllers/register.ts

```diff
@@ -333,9 +333,20 @@ export const processAddServiceForm: RequestHandler = async (req, res, next) => {
 
     req.session.serviceName = req.body.serviceName;
 
-    const generatedClient = await s4.generateClient(service, nonNull(req.session.authenticationResult));
-    const body = JSON.parse(generatedClient.data.output).body;
-    const serviceId = JSON.parse(body).pk;
+    let serviceId = "";
+    let body;
+
+    try {
+        console.info("Generating Client to Service:" + service.serviceName);
+        const generatedClient = await s4.generateClient(service, nonNull(req.session.authenticationResult));
+        body = JSON.parse(generatedClient.data.output).body;
+        serviceId = JSON.parse(body).pk;
+    } catch (error) {
+        console.error("Unable to Register Client to Service - Service Items removed");
+        console.error(error);
+        await s4.deleteServiceEntries(uuid);
+        return res.render("there-is-a-problem.njk");
```

Comment:
return res.render("rollback-service-entries.njk");

Reply:
Hi Rafal, I think the actual solution is not what you suggested but to remove the rollbackServiceEntriesForClient Request Handler and Routes associated with it - I think this was originally put in to do the Logic of removing the Entries but resulted in just redirecting to the Problem Page which I now do in the Catch Section anyway (the Handler is effectively redundant).

Reply:
Rafal has agreed to removal of redundant Request Handler and associated Route.